numpy/core/src/umath/override.c

```diff
@@ -554,14 +560,15 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
             goto fail;
         }
 
-        /* Set the self argument, since we have an unbound method */
-        Py_INCREF(override_obj);
+        /*
+         * Set the self argument of our unbound method.
+         * This also steals the reference, so no need to DECREF after.
+         */
         PyTuple_SetItem(override_args, 0, override_obj);
-
         /* Call the method */
         *result = PyObject_Call(
             override_array_ufunc, override_args, normal_kwds);
-
+        Py_DECREF(override_array_ufunc);
```

Comment:
Hopefully I got them all -- would it be possible to have, at the end of the whole test suite, a forced garbage collection and then a check whether anything in the environment survives?